src/simulate/event_transformer/Transformer.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import utils
+import random
+import json
+
+
+class Transformer:
+
+    USER_STR = "https://mcommunity.umich.edu/#profile:"
+
+    def __init__(self, eventData, config_data):
+        self.eventData = eventData
+        self.envelope = eventData
+        self.event = self.eventData['data'][0]
+        self.config_data = config_data
+
+    def change_user_in_event(self):
+        try:
+            sample_users = self.config_data[utils.PROPS_USERS]
+        except KeyError:
+            logging.error("configuration yaml is missing the \"users\" key")
+            return
+
+        if sample_users is None:
+            logging.error("No Users information added to the configuration yml file")
+            return
+        user = random.choice(sample_users)
+        user_id = self.USER_STR + user[utils.PROPS_USER_ID]
+        user_sis_id = user[utils.PROPS_USER_SIS_ID]
+        user_canvas_id = user[utils.PROPS_USER_CANVAS_ID]
+        self.event['actor'] = user_id
+        self.event['session']['user']= user_id
+        self.event['membership']['member'] = user_id
+        # federated user info
+        fed_session = self.event['federatedSession']
+        fed_msg_params = fed_session['messageParameters']
+        fed_session['user'] = user_id
+        fed_msg_params['custom_canvas_user_id'] = user_canvas_id
+        fed_msg_params['lis_person_sourcedid'] = user_sis_id
+        return user
+
+    def change_course_info(self,user):
+        action = self.event['action']
+        try:
+            sample_courses = self.config_data[utils.PROPS_COURSES]
+        except KeyError:
+            logging.error("configuration yaml is missing the \"courses\" key")
+            return
+
+        if sample_courses is None:
+            logging.error("No courses information added to the configuration yml file")
+            return
+
+        a_course_info = random.choice(sample_courses)
+        # targeted to Media events
+        if action != 'LoggedIn':
+            logging.debug("Action=%s", action)
+            object = self.event['object']
+            target = self.event['target']
+            object['name']="Lecture recorded on "+utils.get_current_date()
```

Comment:
This doesn't need to be randomized.  The time at which a student **_watches_** a lecture needs to be randomized, but not the time at which the lecture was _recorded_.

Reply:
As per this comment, https://github.com/tl-its-umich-edu/caliper-event-analysis/pull/14#discussion_r147217156 asked to give a name to the `object['name']` when i looked at the actual Json data all the object names had this string `Lecture recording on 09/23/16`. So i figured a need to randomize the data. If it is not that what is the suggested name?

Reply:
Sorry if I missed understood on this, I thought `dateCreated`, `dateModified`, `startedAtTime` in the various event objects(like `object`,`target` etc) doesn't need to have the current time, I have no problem in doing this. All that i made current is the `sendTime` and `eventTime`